rfcs/text/0015_nodejs_clustering.md

```diff
@@ -0,0 +1,554 @@
+Note that this doesn’t solve the problem for the console appender, which is probably a no-go.
+
+2. Add the process info to the log pattern and output it with the log message
+
+We could add the process name information to the log messages, and add a new conversion to be able to display it with 
+the pattern layout, such as `%worker` for example.
+
+the default pattern could evolve to (ideally, only when clustering is enabled)
+
+```
+[%date][%level][%worker][%logger] %message
+```
+
+The logging output would then look like
+
+```
+[2021-03-02T10:23:41.834+01:00][INFO ][worker-1][plugins-service] Plugin initialization disabled.
+[2021-03-02T10:23:41.840+01:00][INFO ][worker-2][plugins-service] Plugin initialization disabled.
+```
+
+This will require some non-trivial changes in the logging system implementation, as currently the BaseLogger has no logic to 
+enhance the log record with context information before forwarding it to its appenders, but this still
+seems to be the best solution. (This may even be the first step to implement our MDC)
+
+Notes:
+- The coordinator will probably need to output logs too. `%worker` would be interpolated to `coordinator` 
+  for the coordinator process
+- Even if we add the `%worker` pattern, do we want to also allow users to specify per-worker log files? 
+
+### The rolling-file appender
+
+The rolling process of the `rolling-file` appender is going to be problematic in clustered mode, as it will cause 
+concurrency issues during the rolling. We need to find a way to have this rolling stage clustered-proof.
+
+Identified options are:
+
+1. have the rolling file appenders coordinate themselves when rolling
+
+By using a broadcast message based mutex mechanism, the appenders could acquire a ‘lock’ to roll a specific file, and
+notify other workers when the rolling is complete (quite similar to what we want to do with SO migration for example).
+
+An alternative to this option would be to only have the main worker handle the rolling logic. We will lose control
+on the exact size the file is when rolling, as we would need to wait until the main worker receives a log message
+for the rolling appender before the rolling is effectively performed. The upside would be that it reduces the inter-workers
+communication to a notification from the main worker to the others once the rolling is done for them to reopen their
+file handler.
+
+2. have the coordinator process perform the rolling
+
+Another option would be to have the coordinator perform the rotation instead. When a rolling is required, the appender 
+would send a message to the coordinator, which would perform the rolling and notify the workers once the operation is complete. 
+
+Note that this option is even more complicated than the previous one, as it forces to move the rolling implementation 
+outside of the appender, without any significant upsides identified.
+
+3. centralize the logging system in the coordinator
```

Comment:
I think this is going to be our only option if we want to write the logs to a single file or stdout without fear of the messages interleaving.